djangocms_audio/cms_plugins.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+from django.utils.translation import ugettext_lazy as _
+
+from cms.plugin_base import CMSPluginBase
+from cms.plugin_pool import plugin_pool
+
+from . import models
+
+
+class AudioPlayerPlugin(CMSPluginBase):
+    model = models.AudioPlayer
+    name = _('Audio Player')
+    allow_children = True
+    child_classes = ['AudioFilePlugin', 'AudioFolderPlugin']
+
+    fieldsets = [
+        (None, {
+            'fields': (
+                'template',
+                'label',
+            )
+        }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                'is_active',
+                'attributes',
+            )
+        })
+    ]
+
+    def get_render_template(self, context, instance, placeholder):
+        return 'djangocms_audio/{}/player.html'.format(instance.template)
+
+
+class AudioFilePlugin(CMSPluginBase):
+    model = models.AudioFile
+    name = _('File')
+    module = _('Audio player')
+    require_parent = True
+    parent_classes = ['AudioPlayerPlugin']
+
+    fieldsets = [
+        (None, {
+            'fields': (
+                'audio_file',
+                'text_title',
+            )
+        }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                'text_description',
+                'text_transcript',
+                'attributes',
+            )
+        })
+    ]
+
+    def get_render_template(self, context, instance, placeholder):
+        return 'djangocms_audio/{}/file.html'.format(
```

Comment:
calling `get_plugin_instance()` is not very encouraged, in this case this results in a performance hit.
Because this is a child plugin that requires a player as parent, I suggest using a context variable to pass the theme to the children.

Like so:

```
{% with player_theme=instance.template %}
{% for plugin in instance.child_plugin_instances %}
{% render_plugin child %}
{% endfor %}
{% endwith %}
```

Then inside the `get_render_template()` method you can do `context['player_theme']` to get it.